sentinelhub/api/catalog.py

```diff
@@ -17,115 +22,99 @@ class SentinelHubCatalog(SentinelHubService):
     """
 
     @staticmethod
-    def _get_service_url(base_url):
+    def _get_service_url(base_url: str) -> str:
         """Provides URL to Catalog API"""
         return f"{base_url}/api/v1/catalog"
 
-    def get_info(self):
+    def get_info(self) -> JsonDict:
         """Provides the main information that define Sentinel Hub Catalog API
 
         `Catalog API reference <https://docs.sentinel-hub.com/api/latest/reference/#operation/getLandingPage>`__
 
         :return: A service payload with information
-        :rtype: dict
         """
-        return self.client.get_json(self.service_url)
+        url = self.service_url
+        return self.client.get_json_dict(url)
 
-    def get_conformance(self):
+    def get_conformance(self) -> JsonDict:
         """Get information about specifications that this API conforms to
 
         `Catalog API reference
         <https://docs.sentinel-hub.com/api/latest/reference/#operation/getConformanceDeclaration>`__
 
         :return: A service payload with information
-        :rtype: dict
         """
         url = f"{self.service_url}/conformance"
-        return self.client.get_json(url)
+        return self.client.get_json_dict(url)
 
-    def get_collections(self):
+    def get_collections(self) -> List[JsonDict]:
         """Provides a list of collections that are available to a user
 
         `Catalog API reference <https://docs.sentinel-hub.com/api/latest/reference/#operation/getCollections>`__
 
         :return: A list of collections with information
-        :rtype: list(dict)
         """
         url = f"{self.service_url}/collections"
-        return self.client.get_json(url, use_session=True)["collections"]
+        return self.client.get_json_dict(url)["collections"]
```

Comment:
so the issue we have now, is that `extract_key` also checks that whatever was extracted is a `JsonDict`. Otherwise the signature of the method no longer makes sense. But here `"collections"` are a list... Its a bit annoying, and I am not sure what to do. The same applies to many of the suggestions in this MR

Reply:
Ah, then I'll leave it as it is. :see_no_evil: